### PR TITLE
last packages in net.minecraft.client.gui.*

### PR DESCRIPTION
**Pull request by MelanX**
net.minecraft.client.gui.*
net.minecraft.client.gui.spectator.*

### CODE REVIEW

**Comment by MelanX** [comment]
Was in the wrong branch for the changes I made, because of this I force pushed it to revert it... I hoped that you wouldn't see my mistake then :D

File: data/net/minecraft/client/gui/Font.mapping [hunk 1]
```diff
@@ -35,10 +43,21 @@ CLASS net/minecraft/client/gui/Font
 		ARG 5 dropShadow
 		ARG 6 matrix
 		ARG 7 buffer
-		ARG 8 transparency
+		ARG 8 transparent
 		ARG 9 colorBackground
 		ARG 10 packedLight
 		ARG 11 bidiFlag
+	METHOD drawInBatch (Lnet/minecraft/network/chat/Component;FFIZLcom/mojang/math/Matrix4f;Lnet/minecraft/client/renderer/MultiBufferSource;ZII)I
+		ARG 1 component
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/Font.mapping]: This is called `text` in all other overloads as well as alternate `draw` methods in this class. 
```suggestion
		ARG 1 text
```

File: data/net/minecraft/client/gui/Font.mapping [hunk 2]
```diff
@@ -35,10 +43,21 @@ CLASS net/minecraft/client/gui/Font
...
+		ARG 2 x
+		ARG 3 y
+		ARG 4 color
+		ARG 5 dropShadow
+		ARG 6 matrix
+		ARG 7 bugger
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/Font.mapping]: lol
```suggestion
		ARG 7 bufferSource
```
Should also be reflected in overloads for those named `buffer` that are a buffer source.

File: data/net/minecraft/client/gui/Font.mapping [hunk 3]
```diff
@@ -35,10 +43,21 @@ CLASS net/minecraft/client/gui/Font
...
+		ARG 8 transparent
+		ARG 9 colorBackground
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/Font.mapping]: This is minor, but unless there's a good reason not to (i.e. a field name) I would change these to `backgroundColor`.

File: data/net/minecraft/client/gui/Font.mapping [hunk 4]
```diff
@@ -89,30 +108,36 @@ CLASS net/minecraft/client/gui/Font
...
+		ARG 1 poseStack
 		ARG 2 text
 		ARG 3 x
 		ARG 4 y
 		ARG 5 color
+	METHOD drawWordWrap (Lnet/minecraft/network/chat/FormattedText;IIII)V
+		ARG 1 text
+		ARG 2 x
+		ARG 3 y
+		ARG 4 maxLength
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/Font.mapping]: (field name of `StringSplitter$LineBreakFinder`)
```suggestion
		ARG 4 maxWidth
```

File: data/net/minecraft/client/gui/Font.mapping [hunk 5]
```diff
@@ -148,6 +173,9 @@ CLASS net/minecraft/client/gui/Font
 		ARG 8 isTransparent
 		ARG 9 colorBackground
 		ARG 10 packedLight
+	METHOD split (Lnet/minecraft/network/chat/FormattedText;I)Ljava/util/List;
+		ARG 1 text
+		ARG 2 maxLength
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/Font.mapping]: (field name of StringSplitter$LineBreakFinder)
```suggestion
		ARG 2 maxWidth
```

File: data/net/minecraft/client/gui/Gui.mapping [hunk 6]
```diff
@@ -30,42 +36,82 @@ CLASS net/minecraft/client/gui/Gui
 		ARG 1 mountEntity
 	METHOD getVisibleVehicleHeartRows (I)I
 		ARG 1 mountHealth
+	METHOD guessChatUUID (Lnet/minecraft/network/chat/Component;)Ljava/util/UUID;
+		ARG 1 component
+	METHOD handleChat (Lnet/minecraft/network/chat/ChatType;Lnet/minecraft/network/chat/Component;Ljava/util/UUID;)V
+		ARG 1 chatType
+		ARG 2 component
+		ARG 3 id
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/Gui.mapping]: `senderId` or `playerId`

File: data/net/minecraft/client/gui/Gui.mapping [hunk 7]
```diff
@@ -30,42 +36,82 @@ CLASS net/minecraft/client/gui/Gui
...
 	METHOD renderHotbar (FLcom/mojang/blaze3d/vertex/PoseStack;)V
-		ARG 1 partialTicks
-		ARG 2 matrixStack
+		ARG 1 partialTick
+		ARG 2 poseStack
 	METHOD renderJumpMeter (Lcom/mojang/blaze3d/vertex/PoseStack;I)V
-		ARG 1 matrixStack
-		ARG 2 xPosition
+		ARG 1 poseStack
+		ARG 2 xPos
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/Gui.mapping]:
```suggestion
		ARG 2 x
```

File: data/net/minecraft/client/gui/MapRenderer.mapping [hunk 8]
```diff
@@ -1,11 +1,23 @@
 CLASS net/minecraft/client/gui/MapRenderer
+	METHOD <init> (Lnet/minecraft/client/renderer/texture/TextureManager;)V
+		ARG 1 textureManager
+	METHOD render (Lcom/mojang/blaze3d/vertex/PoseStack;Lnet/minecraft/client/renderer/MultiBufferSource;ILnet/minecraft/world/level/saveddata/maps/MapItemSavedData;ZI)V
+		ARG 1 poseStack
+		ARG 2 buffer
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/MapRenderer.mapping]:
```suggestion
		ARG 2 bufferSource
```

File: data/net/minecraft/client/gui/MapRenderer.mapping [hunk 9]
```diff
@@ -1,11 +1,23 @@
...
 	METHOD resetData ()V
 		COMMENT Clears the currently loaded maps and removes their corresponding textures
 	CLASS MapInstance
+		METHOD <init> (Lnet/minecraft/client/gui/MapRenderer;ILnet/minecraft/world/level/saveddata/maps/MapItemSavedData;)V
+			ARG 2 id
+			ARG 3 data
 		METHOD draw (Lcom/mojang/blaze3d/vertex/PoseStack;Lnet/minecraft/client/renderer/MultiBufferSource;ZI)V
-			ARG 1 matrixStack
+			ARG 1 poseStack
 			ARG 2 buffer
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/MapRenderer.mapping]:
```suggestion
			ARG 2 bufferSource
```

File: data/net/minecraft/client/gui/spectator/categories/TeleportToPlayerMenuCategory.mapping [hunk 10]
```diff
@@ -1,3 +1,5 @@
 CLASS net/minecraft/client/gui/spectator/categories/TeleportToPlayerMenuCategory
+	METHOD <init> (Ljava/util/Collection;)V
+		ARG 1 playerInfos
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/spectator/categories/TeleportToPlayerMenuCategory.mapping]:
```suggestion
		ARG 1 players
```

**Review by Harleyoc1 — CHANGES_REQUESTED** [review]

**Review by Harleyoc1 — APPROVED** [review]

**Review by sciwhiz12 — APPROVED** [review]

**Pull request merged**